### ISSUE TITLE
ArgoCD CLI - better client/server error reporting

### ISSUE THREAD

**Issue by bkuzmic** [argoproj/argo-cd#17162]
# Summary
Update ArgoCD CLI to return more descriptive error messages

# Motivation
We use ArgoCD CLI (version 2.8.5) in CD pipeline. Often the CLI returns 502 or other 50x messages without any other description. it is hard to trace the error unless inspecting the ArgoCD server logs or going to the ArgoCD UI and trying to manually repeat the action. In this case UI might offer a more explanation of the root cause.
We mostly deploy Helm charts with ArgoCD in the following order:

1. argocd app create --grpc-web ...
2. argocd app set --grpc-web ...
3. argocd app sync --grpc-web ...
4. argocd app wait --grpc-web ...

Here is an example of the response:
`Error  level=fatal msg="rpc error: code = Unknown desc = POST https://argocd.x.x.x/application.ApplicationService/Create failed with status code 502" `

# Proposal
Return more descriptive error messages. In case of Helm chart validation errors, all of these could be returned.

**Comment by rmechi** [comment]
in our case its throwing 502 when 
1) argo not render `helm template .`
2) when argo do not find the branch specified in `Application`

**Comment by bkuzmic** [comment]
@rmechi yes, the Helm chart was not valid. But recently we have Permission Denied errors and the app is not synced correctly. We don't know what is causeing this

**Comment by valkiriaaquatica** [comment]
FYI in my case I had this log in the argocd-server-12123

> ="finished unary call with code InvalidArgument" error="rpc error: code = InvalidArgument desc = application spec for harbor is invalid: InvalidSpecError: Unable to generate manifests in deployments/harbor/argocd: rpc error: code = Unknown de

The problema was I had a typo in the values.yaml  file
